tfe/resource_tfe_variable_test.go

```diff
@@ -189,6 +191,193 @@ func TestAccTFEVariable_update_key_sensitive(t *testing.T) {
+						"tfe_variable.foobar", variable),
+					resource.TestCheckResourceAttr(
+						"tfe_variable.foobar", "value", fmt.Sprintf("%d", variableValue2)),
+					resource.TestCheckResourceAttr(
+						"tfe_variable.foobar", "sensitive", "false"),
+					resource.TestCheckResourceAttr(
+						"tfe_workspace.downstream-readable", "name", fmt.Sprintf("downstream-readable-%d", variableValue2)),
+					resource.TestCheckResourceAttr(
+						"tfe_workspace.downstream-nonreadable", "name", fmt.Sprintf("downstream-nonreadable-%d", variableValue2)),
+				),
+			},
+		},
+	})
+}
+
+func TestAccTFEVariable_varset_readable_value_becomes_sensitive(t *testing.T) {
+	variable := &tfe.VariableSetVariable{}
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+
+	variableValue1 := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+	variableValue2 := variableValue1 + 42
+
+	// Test that if an insensitive variable becomes sensitive, downstream resources depending on the readableValue
+	// may no longer access it, but that the value may still be used directly
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV5ProviderFactories: testAccMuxedProviders,
+		CheckDestroy:             testAccCheckTFEVariableDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTFEVariable_varset_readablevalue(rInt, variableValue1, false),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTFEVariableSetVariableExists(
+						"tfe_variable.foobar", variable),
+					resource.TestCheckResourceAttr(
+						"tfe_variable.foobar", "value", fmt.Sprintf("%d", variableValue1)),
+					resource.TestCheckResourceAttr(
+						"tfe_variable.foobar", "sensitive", "false"),
+					resource.TestCheckResourceAttr(
+						"tfe_workspace.downstream-readable", "name", fmt.Sprintf("downstream-readable-%d", variableValue1)),
+					resource.TestCheckResourceAttr(
+						"tfe_workspace.downstream-nonreadable", "name", fmt.Sprintf("downstream-nonreadable-%d", variableValue1)),
+				),
+			},
+			{
+				Config:      testAccTFEVariable_varset_readablevalue(rInt, variableValue2, true),
+				ExpectError: regexp.MustCompile(`tfe_variable.foobar.readable_value is null`),
+			},
+		},
+	})
+}
+
+func TestAccTFEVariable_readable_value_explicitlySet(t *testing.T) {
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+
+	variableValue1 := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+
+	// Test that we throw an error if the user attempts to explicitly set the readable_value in the configuration
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV5ProviderFactories: testAccMuxedProviders,
+		CheckDestroy:             testAccCheckTFEVariableDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccTFEVariable_readablevalue_setExplicitly(rInt, variableValue1),
+				ExpectError: regexp.MustCompile(`Invalid Configuration for Read-Only Attribute`),
```

Comment:
Given that read-only attributes are a built-in and fairly fundamental feature of the framework, I think we can get away without having this acceptance test! It would be prudent to have if we were using a custom validator, but now that that's out I think we can just remove it. 👍🏼